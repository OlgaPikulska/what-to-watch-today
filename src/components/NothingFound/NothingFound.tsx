import { Box, Button, Typography } from "@mui/material";
import { useRouter } from "next/navigation";
import React from "react";

export const NothingFound = () => {
	const router = useRouter();
	return (
		<Box
			sx={{
				display: "flex",
				flexDirection: "column",
				justifyContent: "center",
				alignItems: "center",
				height: "40vh",
				textAlign: "center",
			}}
		>
			<Typography variant="h6" gutterBottom>
				No movies matching your query were found.
			</Typography>
			<Typography variant="body1" sx={{ marginBottom: 2 }}>
				Please go back to the homepage or try a different search.
			</Typography>
			<Button
				variant="contained"
				onClick={() => router.push("/")}
				size="large"
				sx={{
					color: "#fff",
				}}
			>
				Back to Homepage
			</Button>
		</Box>
	);
};
