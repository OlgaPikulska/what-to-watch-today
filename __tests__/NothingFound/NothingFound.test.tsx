import { NothingFound } from "@/components/NothingFound/NothingFound";
import { render } from "@testing-library/react";
import { useRouter } from "next/navigation";
import { userEvent } from "@testing-library/user-event";

jest.mock("next/navigation", () => ({
	useRouter: jest.fn(),
}));

describe("NothingFound Component", () => {
	const routerPushMock = jest.fn();

	beforeEach(() => {
		(useRouter as jest.Mock).mockReturnValue({ push: routerPushMock });
	});

	afterEach(() => {
		jest.clearAllMocks();
	});

	it("should match the snapshot", () => {
		const { container } = render(<NothingFound />);
		expect(container).toMatchSnapshot();
	});

	it("should render the component correctly", () => {
		const { getByText, getByRole } = render(<NothingFound />);

		expect(getByText("No movies matching your query were found.")).toBeInTheDocument();
		expect(
			getByText("Please go back to the homepage or try a different search."),
		).toBeInTheDocument();

		const button = getByRole("button", { name: "Back to Homepage" });
		expect(button).toBeInTheDocument();
	});

	it("redirects to the homepage when button is clicked", async () => {
		const { getByRole } = render(<NothingFound />);
		const button = getByRole("button", { name: "Back to Homepage" });

		await userEvent.click(button);

		expect(routerPushMock).toHaveBeenCalledTimes(1);
		expect(routerPushMock).toHaveBeenCalledWith("/");
	});
});
