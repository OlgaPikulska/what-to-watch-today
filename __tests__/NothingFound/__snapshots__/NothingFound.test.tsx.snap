// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NothingFound Component should match the snapshot 1`] = `
<div>
  <div
    class="MuiBox-root css-1ntkkrs"
  >
    <h6
      class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-1wmifpn-MuiTypography-root"
    >
      No movies matching your query were found.
    </h6>
    <p
      class="MuiTypography-root MuiTypography-body1 css-9ebprp-MuiTypography-root"
    >
      Please go back to the homepage or try a different search.
    </p>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorPrimary css-152anmu-MuiButtonBase-root-MuiButton-root"
      id=":r0:"
      tabindex="0"
      type="button"
    >
      Back to Homepage
    </button>
  </div>
</div>
`;
